src/Workspaces/Core/Portable/ObsoleteSymbol/AbstractObsoleteSymbolService.cs

```diff
@@ -0,0 +1,176 @@
+
+        foreach (var span in textSpans)
+        {
+            Recurse(span, semanticModel);
+        }
+
+        result.RemoveDuplicates();
+        return result.ToImmutableAndClear();
+
+        void Recurse(TextSpan span, SemanticModel semanticModel)
+        {
+            using var _ = ArrayBuilder<SyntaxNode>.GetInstance(out var stack);
+
+            // Walk through all the nodes in the provided span.  Directly analyze local or parameter declaration.  And
+            // also analyze any identifiers which might be reference to locals or parameters.  Note that we might hit
+            // locals/parameters without any references in the span, or references that don't have the declarations in 
+            // the span
+            stack.Add(root.FindNode(span));
+
+            // Use a stack so we don't blow out the stack with recursion.
+            while (stack.Count > 0)
+            {
+                var current = stack.Last();
+                stack.RemoveLast();
+
+                if (current.Span.IntersectsWith(span))
+                {
+                    var tokenFromNode = ProcessNode(semanticModel, current);
+
+                    foreach (var child in current.ChildNodesAndTokens())
+                    {
+                        if (child.IsNode)
+                        {
+                            stack.Add(child.AsNode()!);
+                        }
+
+                        var token = child.AsToken();
+                        if (token != tokenFromNode)
+                            ProcessToken(semanticModel, child.AsToken());
+
+                        foreach (var trivia in token.LeadingTrivia)
+                        {
+                            if (trivia.HasStructure)
+                            {
+                                stack.Add(trivia.GetStructure()!);
+                            }
+                        }
+
+                        foreach (var trivia in token.TrailingTrivia)
+                        {
+                            if (trivia.HasStructure)
+                            {
+                                stack.Add(trivia.GetStructure()!);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        SyntaxToken ProcessNode(SemanticModel semanticModel, SyntaxNode node)
+        {
+            if (syntaxFacts.IsUsingAliasDirective(node))
+            {
+                syntaxFacts.GetPartsOfUsingAliasDirective(node, out _, out var aliasToken, out var name);
+                if (!aliasToken.Span.IsEmpty)
+                {
+                    // Use 'name.Parent' because VB can't resolve the declared symbol directly from 'node'
+                    var symbol = semanticModel.GetDeclaredSymbol(name.GetRequiredParent(), cancellationToken);
+                    if (IsSymbolObsolete(symbol))
+                        result.Add(aliasToken.Span);
+                }
+
+                return aliasToken;
```

Comment:
Cleaner (i think) is that you should be able to do GetDeclaredSymbol on the node to get the IAliasSymbol, and then just check the Target of that symbol.  this def should work for C#... and should be good for VB i think.

Reply:
I wasn't quite sure what the suggestion here was.